nixos/tests/clatd.nix

```diff
@@ -102,6 +103,37 @@ import ./make-test-python.nix ({ pkgs, lib, ... }:
         };
       };
 
+      systemd.network.networks."40-eth2" = {
+        networkConfig.IPv6SendRA = true;
+        ipv6Prefixes = [ { Prefix = "2001:db8::/64"; } ];
+        ipv6PREF64Prefixes = [ { Prefix = "64:ff9b::/96"; } ];
+        ipv6SendRAConfig = {
+          EmitDNS = true;
+          DNS = "_link_local";
+        };
+      };
+
+      services.resolved.extraConfig = ''
```

Comment:
So resolved does not have any support for this yet? Is there an upstream issue we can link to?

I assume once that's supported, we would not need to do any custom DNS setup here?

Reply:
This config is just set as to not conflict with the coredns instance running. clatd will fetch the PLAT prefix via DNS64, which is served up by coredns in the configuration I made.

Reply:
Ah I see what you're saying, does resolved have support for serving up DNS64 synthesized records. AFAIK no it does not, and I don't see any issue upstream.

Reply:
Yes, exactly. Would you feel like opening one?

- This feels like a logical successor after the pref64 stuff, so it would make sense to signal interest in it
- It would allow others looking at the code and module options to understand why we went with the coredns route 
- it makes it easy to know when we can drop coredns from here again

Reply:
I'm actually not super sure that adding a DNS64 component to systemd-resolved would be a good idea. If a client running systemd-networkd is already able to obtain the PREF64 prefix from the router advertisements, what good is it for systemd-resolved to serve up modified DNS responses to applications when the machine should be smart enough to setup a clat interface so the unmodified DNS A RRs can be used directly. I do get that we are kinda in a weird place right now where systemd-networkd exposes this information about the PLAT prefix existing, but doesn't (yet) take any responsibility in setting up a clat interface, and at the same time no software that I'm aware yet reads this exposed information, so software such as clatd is left to do "discovery" of the presence of a NAT64 machine by virtue of it receiving DNS64 responses. I think the logical move forward is for systemd-networkd to add proper 464xlat support and/or for more software to start reading from the info exposed by networkd.

In regards to using coredns for this test, I'm happy to use another DNS server if you think that would be best, I am just personally familiar with the coredns config and with the config being quite terse for this use case.

Reply:
Relevant systemd issue: https://github.com/systemd/systemd/issues/23674

Reply:
Ok yeah, fair points. I guess we can generally follow this as a tracking issue. Thanks for the digging!